Update Sample File Upload and upgrade buttons to Shadcn button

nithish1018 opened the pull request:
## Proposed Changes

- Fixes #9153 
- Updated File Upload process to match with existing FileUpload
- Updated regular Button and ButtonV2 to Shadcn button

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new file upload mechanism in the UpdateStatusDialog, enhancing user experience with improved error handling and progress indication.
	- Added a new label for report uploads in the localization files.
	- Introduced a new `primary` button variant for enhanced visual options.

- **Bug Fixes**
	- Restructured file upload handling to ensure smoother functionality and user interaction.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nithish1018 committed:
Added Clear Image functionality

nithish1018 committed:
removed unneccessary line

nithish1018 committed:
Added bot suggestions

nithishsiliveru commented on src/components/Patient/UpdateStatusDialog.tsx:
```diff
@@ -236,10 +236,17 @@ const UpdateStatusDialog = (props: Props) => {
                   type="submit"
                   onClick={handleUpload}
-                  disabled={uploadDone}
+                  disabled={file ? false : true}
```
Done

shivankacker reviewed (changes requested):
@nithish1018 we need to migrate the entire flow to use the new `useFileUpload` and `useFileManager` hooks. Look at consent and file upload component for reference.

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Fixed search icon overlay

nithish1018 committed:
removing unrelated change

nithish1018 committed:
Updated file upload with useFileUpload hook

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Added types for sample

nithish1018 commented:
> check the code rabbit suggestion

Done

nithishsiliveru commented on src/components/Patient/UpdateStatusDialog.tsx:
```diff
@@ -10,31 +10,24 @@ import CheckBoxFormField from "@/components/Form/FormFields/CheckBoxFormField";
-  sample: SampleTestModel;
-  handleOk: (sample: SampleTestModel, status: number, result: number) => void;
+  sample: any;
+  handleOk: (sample: any, status: number, result: number) => void;
```
Added, thank you for the suggestion

shivankacker committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
\Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nihal467 commented:
@nithish1018 The design looks inconsistent compared to our file upload component in the screenshot below. Please implement a similar design for consistency. 

![image](https://github.com/user-attachments/assets/9d969df7-468b-4187-a433-46a2adb2c0d4)

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Updated the file upload UI

nithish1018 commented:
> @nithish1018 The design looks inconsistent compared to our file upload component in the screenshot below. Please implement a similar design for consistency.
> 
> ![image](https://private-user-images.githubusercontent.com/57055998/390595098-9d969df7-468b-4187-a433-46a2adb2c0d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3Njc1MTAsIm5iZiI6MTczMjc2NzIxMCwicGF0aCI6Ii81NzA1NTk5OC8zOTA1OTUwOTgtOWQ5NjlkZjctNDY4Yi00MTg3LWE0MzMtNDZhMmFkYjJjMGQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDA0MTMzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2NTk2ZmNjZGFmNmE2NTg2OTA2ZjVjZmMyMzVkYTRkYTAyNWVkMmIxMmJmYjQ1MWUyNWUyZTEwZTBlMTY2MWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.s3qk_IM8TgVjJjENXp9zZCx61kHzC8R2l14RSa5B3LE)

I made the requested changes, please review it

nithishsiliveru commented on src/components/Patient/UpdateStatusDialog.tsx:
```diff
@@ -218,36 +176,78 @@ const UpdateStatusDialog = (props: Props) => {
+                {uploadButtons
+                  .filter((b) => b.show !== false)
+                  .map((button, i) => (
+                    <label
+                      key={i}
+                      className="flex w-full cursor-pointer items-center justify-center gap-2 rounded-lg border border-dashed border-primary-500/20 bg-primary-500/10 p-3 text-primary-700 transition-all hover:bg-primary-500/20 md:p-6"
+                      onClick={button.onClick}
+                      id={button.id}
+                    >
+                      <CareIcon icon={button.icon} className="text-2xl" />
+                      <div className="text-lg">{button.name}</div>
+                      {button.children}
+                    </label>
```
The existing FileUpload component doesn't have such implementation

Jacobjeevan commented on src/components/Patient/UpdateStatusDialog.tsx:
```diff
@@ -218,36 +176,78 @@ const UpdateStatusDialog = (props: Props) => {
+                  required
+                />
+                <div className="flex items-center gap-2">
+                  <ButtonV2
```
Replace instances of button and ButtonV2 with shadcn's Button (from components/ui); we are moving to using shadcn for newer items 👍

nithishsiliveru replied:
Sure, will update it

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Updated Button to shadcn button

nithish1018 committed:
removed unnecessary line

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nihal467 commented:
![image](https://github.com/user-attachments/assets/fe793ad5-aaf2-4519-b6d4-5cbb5db18405)

- [ ] only keep the discard button on the bottom and remove the other button

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
removed remove icon in favour of discard button

nithish1018 commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/391199401-fe793ad5-aaf2-4519-b6d4-5cbb5db18405.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5MzI5NjYsIm5iZiI6MTczMjkzMjY2NiwicGF0aCI6Ii81NzA1NTk5OC8zOTExOTk0MDEtZmU3OTNhZDUtYWFmMi00NTE5LWI2ZDQtNWNiYjVkYjE4NDA1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTMwVDAyMTEwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3MWJmNWI1MDM1Y2NlNzI4Njg0MWQ0OTYwMjVlMWRmNWFiYzNkMzVlNTQzYWRiMTg2YjhmZThjNWFiZGNiMTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tCLvnNu8Rrus_qck1hVS50VY7FmFc1nx4c5OL15tYug)
> 
> * [ ]  Only keep the discard button at the bottom and remove the other button next to the file name to maintain consistency with the existing file upload.
> 
> @nithish1018 make sure to do a basic QA before requesting a review

Actually the existing File Upload has that extra option to clear image , so I've added that here aswell.

![image](https://github.com/user-attachments/assets/fb79e1ca-2bd0-4256-9e30-18c5d0165e1d)

Now I've updated the PR as per your suggestions, please review it

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

rithviknishad commented on src/components/ui/button.tsx:
```diff
@@ -15,6 +15,8 @@ const buttonVariants = cva(
         outline:
           "border border-gray-200 bg-white shadow-sm hover:bg-gray-100 hover:text-gray-900 dark:border-gray-800 dark:bg-gray-950 dark:hover:bg-gray-800 dark:hover:text-gray-50",
+        primary:
+          "bg-primary-500 text-white shadow hover:bg-primary-700/90 dark:bg-primary-100 dark:text-primary-900 dark:hover:bg-primary-100/90",
```
As per figma designs, primary variant would be using shade 700.

<img width="319" alt="image" src="https://github.com/user-attachments/assets/04613e9e-b32b-49bc-89a3-4c68be5059bc">

https://www.figma.com/design/Z93EYKSa1MdBmXndsMBJSQ/Care?node-id=1285-2299&node-type=canvas&t=li0D8fzrfJMTag7g-0

```suggestion
          "bg-primary-700 text-white shadow hover:bg-primary-700/90 dark:bg-primary-100 dark:text-primary-900 dark:hover:bg-primary-100/90",
```

nithishsiliveru replied:
I'll update it

nithish1018 committed:
updated primary button color

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
fixed eslint issues

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nithish1018 committed:
Merge branch 'develop' into issues/9153/addclearimage

nihal467 commented:
LGTM

rithviknishad commented on src/components/Patient/UpdateStatusDialog.tsx:
```diff
@@ -65,11 +61,27 @@ const UpdateStatusDialog = (props: Props) => {
+    allowedExtensions: ["pdf", "jpg", "jpeg", "png"],
+    allowNameFallback: true,
+  });
+  const uploadButtons: {
+    name: string;
+    icon: IconName;
+    onClick?: () => void;
+    children?: ReactNode;
+    show?: boolean;
+    id: string;
+  }[] = [
+    {
+      name: t("choose_file"),
+      icon: "l-file-upload-alt",
+      children: <fileUpload.Input />,
+      id: "upload-file",
+    },
+  ];
```
I don't see any places where the `uploadButtons` array is mutated. And it only contains 1 element. Why is it an array? Can't we just directly inline the component?

nithishsiliveru replied:
Actually I tried to rewrite this component just like the  File Upload component, Now I've pushed the requested changes. Please update the review

nithish1018 committed:
removed button array
